python/lsst/pipe/tasks/makeCoaddTempExp.py

```diff
@@ -875,3 +881,37 @@ def prepareCalibratedExposures(self, calExpList, backgroundList=None, skyCorrLis
             # Apply skycorr
             if self.config.doApplySkyCorr:
                 mi -= skyCorr.getImage()
+
+
+def reorderInputRefs(inputRefs, outputKeyOrder, dataIdKey):
+    """Reorder inputRefs per outputKeyOrder
+
+    Any inputRefs which are lists will be resorted per specified key e.g.,
+    detector. Only iterables will be reordered, and values can be of type
+    `lsst.pipe.base.connections.DeferredDatasetRef` or
+    `lsst.daf.butler.core.datasets.ref.DatasetRef`
+
+    Parameters
+    ----------
+    inputRefs : `lsst.pipe.base.connections.InputQuantizedConnection`
+        Input references to be reordered and padded.
+    outputKeyOrder : iterable
+        Iterable of values to be compared with inputRef's dataIdKey.
+        Must be hashable
+    dataIdKey :  `str`
+        dataIdKey in the dataRefs to compare with the outputKeyOrder.
+
+    Returns:
+    --------
+    inputRefs: `lsst.pipe.base.connections.InputQuantizedConnection`
+        Sorted inputRefs.
+    """
+    for key, refs in inputRefs:
+        if isinstance(refs, Iterable):
+            if hasattr(refs[0], "dataId"):
+                inputKeyOrder = [ref.dataId[dataIdKey] for ref in refs]
+            else:
+                inputKeyOrder = [ref.datasetRef.dataId[dataIdKey] for ref in refs]
+            if inputKeyOrder != outputKeyOrder:
+                setattr(inputRefs, key, reorderAndPadList(refs, inputKeyOrder, outputKeyOrder))
```

Comment:
Can this be `inputRefs[key] = reorderAndPadList(refs, inputKeyOrder, outputKeyOrder)`?

Reply:
Nope.
```(Pdb) inputRefs[key] 
*** TypeError: 'InputQuantizedConnection' object is not subscriptable
```